### PR TITLE
PUPPETDB SD: Validate HTTP config

### PR DESCRIPTION
**Pull request by slashpai**
Related-to #12810

### CODE REVIEW

**Review by roidelapluie — APPROVED** [review]
Tested & LGTM

**Comment by roidelapluie** [comment]
maybe add a job_name so it does not error on main

File: config/testdata/puppetdb_token_file.bad.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+scrape_configs:
+  - puppetdb_sd_configs:
```
[the remark this answers is not in the document]
Comment (slashpai) [reply config/testdata/puppetdb_token_file.bad.yml:2]: sure, I missed the job name

**Comment by slashpai** [comment]
Added job name

**Comment by roidelapluie** [comment]
Thanks!

**Pull request merged**